source/common/http/conn_manager_impl.h

```diff
@@ -310,12 +317,13 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void blockRouteCache();
     // Return true if the cached route is blocked.
     bool routeCacheBlocked() const {
-      ENVOY_BUG(!route_cache_blocked_,
-                "Should never try to refresh or clear the route cache when "
-                "it is blocked! To temporarily ignore this new constraint, "
-                "set runtime flag "
+      ENVOY_LOG(warn,
+                "Route cache will be blocked after response headers are sent and should never "
+                "try to refresh or clear the route cache when it is blocked! To temporarily ignore "
+                "this new constraint, set runtime flag "
                 "`envoy.reloadable_features.prohibit_route_refresh_after_response_headers_sent` "
                 "to `false`");
```

Comment:
This seems like this could log spam really badly. Can you can summarize what is leading to this change? The other issue has a lot of comments in it. Thank you.

/wait

Reply:
Add all related context in the PR's description. And If you think this will introduce lots of unnecessary noises, may be we can change ENVOY_LOG to ENVOY_LOG_FIRST_N.

Reply:
OK makes sense. It seems like (1) above is a bug fix. Given the fix, can't we keep ENVOY_BUG? Is there any reason it should happen with the fix?

Reply:
no strong reason. Just because I am worried about others potential corner cases that not be covered and think ENVOY_BUG is too strict.
Although keeping ENVOY_BUG also ok to me.

Reply:
ENVOY_BUG compiles out in release builds, and if it hits, it feels like a real bug to me, so I would leave it?